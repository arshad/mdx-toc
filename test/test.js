const fs = require("fs");
const path = require("path");
const remark = require("remark");
const mdx = require("remark-mdx");
const frontmatter = require("remark-frontmatter");
const toc = require("..");

const doc = fs.readFileSync(path.join(__dirname, "./fixture.mdx"), "utf8");

it("extracts toc from mdx", () => {
  const result = remark().use(frontmatter).use(mdx).use(toc).processSync(doc);

  expect(result.data).toMatchSnapshot();
});
