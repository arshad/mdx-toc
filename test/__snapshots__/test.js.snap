// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extracts toc from mdx 1`] = `
Array [
  Object {
    "depth": 1,
    "value": "Section heading 1",
  },
  Object {
    "depth": 2,
    "value": "Section heading 2",
  },
  Object {
    "depth": 2,
    "value": "Another heading 2",
  },
  Object {
    "depth": 3,
    "value": "Section Heading 3",
  },
]
`;
